### ISSUE TITLE
Wrong signature for axpy blas functions

### ISSUE THREAD

**Issue by finkandreas** [kokkos/kokkos-kernels#329]
In the file `src/impl/tpls/KokkosBlas1_axpby_tpl_spec_decl.hpp` the function axpy is declared as
```
extern "C" void daxpy_( const int* N, const double* alpha,
                                      const double* x, const int* x_inc,
                                      const double* y, const int* y_inc);
```
but it should be:
```
extern "C" void daxpy_( const int* N, const double* alpha,
                                      const double* x, const int* x_inc,
                                      double* y, const int* y_inc);
```
The second last argument is an output argument, so it shall not be const. This applies also to the other axpy functions.

**Comment by mhoemmen** [comment]
Fix in PR #329.  btw @finkandreas GitHub lets you submit a PR with just a few clicks -- you can edit the file in the browser -- so please feel free to submit a fix for issues like this if you like :-)

**Comment by ndellingwood** [comment]
PR #329 with fix by @mhoemmen merged.
